lib/services/post.ts

```diff
@@ -0,0 +1,155 @@
+import { Post, PostStatus, User } from "@prisma/client";
+import { prisma } from "./db";
+import { deleteImage, uploadImage } from "./image";
+
+export type PostWithUser = Post & {
+  user: User;
+};
+
+export type CreatablePost = {
+  price: number;
+  description: string;
+  image: Buffer;
```

Comment:
Using a Buffer instead of a URL here differs from the spec. I don't necessarily disagree with the choice but could you update the spec in that case then? Also, should the image be optional?

Reply:
I guess we can make the image optional. I will update the spec once the spec PR is merged

Reply:
Spec updated. I also updated the image service docs to say "Buffer" instead of "Blob"